src/main/java/net/spy/memcached/ops/MonitoringOperationStateChangeObserver.java

```diff
@@ -0,0 +1,32 @@
+package net.spy.memcached.ops;
+
+import net.spy.memcached.metrics.MetricCollector;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An {@link OperationStateChangeObserver} implementation for monitoring purposes.
+ * */
+public class MonitoringOperationStateChangeObserver implements OperationStateChangeObserver {
```

Comment:
Must we create a new observer instance on each operation? There'll be A LOT of instances in the heap

Reply:
Nope, changed that.
In short: 
Now there is a single stateless listener - which will get the needed stuff when called, as method params.
+
I've also cached the final metrics names strings on my own - which is ugly because the build is with java 6 - didn't wanna change that or add a dependency on other external module....please take a look.